Earn Passive Income with AMM (#2605)
XRPLF/xrpl-dev-portal: docs/tutorials/javascript/trade-on-ledger/earn-passive-income-as-a-liquidity-provider.md

## Steps

### 1. Connect to the network and generate credentials

{% code-snippet file="/_code-samples/add-amm-lp/js/add-amm-lp.js" language="js" from="// Define the network client" before="// Acquire TST tokens" /%}

### 2. Acquire TST tokens

{% code-snippet file="/_code-samples/add-amm-lp/js/add-amm-lp.js" language="js" from="// Acquire TST tokens" before="// Deposit assets into AMM" /%}

### 3. Deposit assets to the AMM

You can contribute either one or both assets to an AMM pool. Contributing just one asset can be easier, but incurs a fee that is debited from the LP Tokens paid out. If you contribute both assets in the pool, you aren't charged a fee.

{% code-snippet file="/_code-samples/add-amm-lp/js/add-amm-lp.js" language="js" from="// Deposit assets into AMM" before="// Vote on fees" /%}

### 4. Vote on trading fees

This step is optional, but you're encouraged to submit a vote for a fee structure you believe makes sense.

{% code-snippet file="/_code-samples/add-amm-lp/js/add-amm-lp.js" language="js" from="// Vote on fees" before="// Get LP tokens value" /%}

> ### 5. Check the value of your LP tokens

Reviewer: I still think it makes more sense to explicitly call out "the part where you just sit and wait" for an indeterminate amount of time, since the code sample doesn't do that but in a realistic situation you would, so that people know where to break.

```suggestion
### 5. Wait for your LP Tokens to accrue value
```
